src/gui/widgets/backupwidget.cpp

```diff
@@ -43,28 +46,66 @@ BackupWidget::~BackupWidget()
 }
 
 
-void BackupWidget::on_tableView_clicked(const QModelIndex &index) {
-    selected = model->item(index.row(), 0);
-    DEB << "Item at row:" << index.row() << "->" << selected->data(Qt::DisplayRole);
+void BackupWidget::on_tableView_clicked(const QModelIndex &index)
+{
+    selectedBackupName = model->item(index.row(), 0);
+    DEB << "Item at row:" << index.row() << "->" << selectedBackupName->data(Qt::DisplayRole);
 }
 
 void BackupWidget::on_createLocalPushButton_clicked()
 {
-    // Copy database to backupdir
+    NOT_IMPLEMENTED
+    QString filename = QFileDialog::getSaveFileName(
+                this,
+                "Choose destination file",
+                AStandardPaths::directory(AStandardPaths::Backup).absolutePath(),
+                "*.db"
+    );
+
+    if(filename.endsWith(".db") == false) {  // [G]: clunky im sure it can be enforced by QFileDialog
+        filename.append(".db");
+    }
+
+    if(aDB->createBackup(filename) == false) {
+        WARN << "Could not create local file:" << filename;
+    }
+
+    // [G] TODO: propably make a function out of this for future tweaks
+    QFileIconProvider provider;
+    QMap<QString, QString> summary = aDB->databaseSummary(filename);
+    model->appendRow({new QStandardItem(provider.icon(QFileIconProvider::File), QFileInfo(filename).baseName()),
+                      new QStandardItem(summary["total_flights"]),
+                      new QStandardItem(summary["total_tails"]),
+                      new QStandardItem(summary["total_pilots"]),
+                      new QStandardItem(summary["max_doft"]),
+                      new QStandardItem(summary["total_time"])
+                     });
 }
 
 void BackupWidget::on_restoreLocalPushButton_clicked()
 {
-    // Restore the selected entry from the list
+    if(selectedBackupName == nullptr) {
+        INFO << "No backup selected";
+        return;
+    }
+    QString backup_name = selectedBackupName->data(Qt::DisplayRole).toString();
+    if(aDB->restoreBackup(backup_name) == false) {
+        WARN << "Couldnt restore" << backup_name;
+    }
 }
 
 void BackupWidget::on_deleteSelectedPushButton_clicked()
 {
-    DEB << "deleting:" << selected->data(Qt::DisplayRole);
+    if(selectedBackupName == nullptr) {
+        INFO << "No backup was selected";
+        return;
+    }
+    DEB << "deleting:" << selectedBackupName->data(Qt::DisplayRole);
 }
 
 void BackupWidget::on_createExternalPushButton_clicked()
```

Comment:
Here I wouldn't open `AStandardPaths::Backup` since this is used to create an external backup to a custom location specified by the user. The Backupdir is for use with internal backups (the ones that get displayed in the tableview), and the user should not put his 'external' backups in this folder, so I would opt for opening the QFileDialog at `QStandardPaths::HomeLocation`

Reply:
WIll implement!